core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -5731,7 +5731,7 @@ public final void updateRoleName(String roleName, String newRoleName) throws Use
             throw new UserStoreException(errorCode + " - " + errorMessage);
         }
 
-        if (isExistingRole(newRoleName)) {
+        if (!StringUtils.equalsIgnoreCase(roleName, newRoleName) && isExistingRole(newRoleName)) {
```

Comment:
`isExistingRole ` Hope this method does not ignore case?

Reply:
MSSQL is case insensitive. Therefore in  `isExistingRole` method, it will not consider the case insensitivity in the DB check. So if that condition is only checked, it will throw an exception. That's why `!StringUtils.equalsIgnoreCase(roleName, newRoleName)` condition has been added to check whether this is a case sensitive update.

Reply:
Understood, thanks! 

Can you just check whether is it possible for two different roles to exist with case-sensitive names?

Ex: `testrole` and `testRole`.  If possible then this additional condition might break non ms-sql DBs if someone tries to change the name of `testrole` to `testRole`

Reply:
It's not possible to add two roles like `testrole` and `testRole` from the console or the management console.

Reply:
Is that validation is enforced at the same level? (abstract userstore manager)

The ideal solution is to somehow find a way to fix `isExistingRole` to work with MsSQL with case-sensitive check IMO. Because we are `TRUSTING` that case sensitive apps wont get added. If someone else added that capability later he may not aware about this logic. So shall we try to fix mssql level? but if that is not possible,

then lets use this fix and write a unit/integration test(if not already exists),

1. Create a role `test-role`
2. Try to create `test-Role`
3. assert the failure. There we can put a code level comment there and say this behaviour is expected this check in the abstract userstore manager also.

Also, it would be good to add a code level comment here saying something like `it's not possible to add two case-sensitively different roles from other APIs, therefore the possibility to have an existing role in the `newRoleName `'s name is zero. Because of that there will not be an error `

Reply:
The same `isExistingRole` logic has been used for adding role as well. Therefore IMO it would be better to do this validation at this point. As you suggested, code level comment was added to clarifications and added integration tests for adding role with case sensitive role name to ensure that case sensitive group names will not be added.